src/Centreon/Domain/Monitoring/Model/ResourceDetailsTrait.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/*
+ * Copyright 2005 - 2020 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Domain\Monitoring\Model;
+
+use Centreon\Domain\Monitoring\Resource;
+use Centreon\Domain\Monitoring\ResourceStatus;
+use Centreon\Domain\Acknowledgement\Acknowledgement;
+use CentreonDuration;
+
+/**
+ * The model enrich the Host model
+ */
+trait ResourceDetailsTrait
+{
+    /**
+     * @var Resource|null
+     */
+    protected $parent;
+
+    /**
+     * @var ResourceStatus|null
+     */
+    protected $status;
+
+    /**
+     * @var string|null
+     */
+    protected $tries;
+
+    /**
+     * @return \Centreon\Domain\Downtime\Downtime[]
+     */
+    abstract public function getDowntimes(): array;
+
+    /**
+     * @param Acknowledgement|null $acknowledgement
+     * @return Host
+     */
+    abstract public function getAcknowledgement(): ?Acknowledgement;
+
+    /**
+     * @return string|null
+     */
+    public function getDowntimesAuthor(): ?string
+    {
+        $downtimes = $this->getDowntimes();
+        return $downtimes
+            ? (end($downtimes))->getAuthorName()
```

Comment:
the return downtimes should be a list, so we should return author by downtime (no the last one)

Reply:
it's changed right now to return the parent list of downtimes and acknowledgment